base/arrayshow.jl

```diff
@@ -320,6 +320,7 @@ print_array(io::IO, X::AbstractArray) = show_nd(io, X, print_matrix, true)
 # typeinfo aware
 # implements: show(io::IO, ::MIME"text/plain", X::AbstractArray)
 function show(io::IO, ::MIME"text/plain", X::AbstractArray)
+    isempty(X) && ndims(X) == 1 && return show(io, X)
```

Comment:
This is essentially what I suggested in #34733, but why not `get(io, :compact, false) === true && return show(io, X)`?

Reply:
Hmm, there are tests for this, e.g.
```
replstr([x, x], :compact => true) == "2-element Array{Float64,1}:\n 3.14159\n 3.14159"
```
I guess the idea was to be able to keep the overall repl printing format but forcing elements to be shown compactly.

Reply:
I see. I think this is another instance of the limitation for using `:compact` to mean "small width" and "single line."